share/spice/js_keycodes/js_keycodes.js

```diff
@@ -0,0 +1,151 @@
+        { key: 'f4', code: 115 },   { key: 'f5', code: 116 },   { key: 'f6', code: 117 },
+        { key: 'f7', code: 118 },   { key: 'f8', code: 119 },   { key: 'f9', code: 120 },
+        { key: 'f10', code: 121 },  { key: 'f11', code: 122 },  { key: 'f12', code: 123 },
+
+        { key: 'num lock', code: 144 }, { key: 'scroll lock', code: 145 },  { key: ';', code: 186 },
+        { key: '*', code: 106 },        { key: '-', code: 189 },            { key: '.', code: 190 },
+        { key: "=", code: 187 },        { key: ',', code: 188 },            { key: '/', code: 191 },
+        { key: '\\', code: 220 },       { key: '(', code: 219 },            { key: ')', code: 221 },
+        { key: 'quote', code: 222 }
+    ];
+
+    function getKeycodeByKey(key) {
+        return $.grep(keycodeLookup, function (keycodeItem) {
+            return keycodeItem.key === key;
+        })[0];
+    }
+
+    function getKeycodeByCode(code) {
+        return $.grep(keycodeLookup, function (keycodeItem) {
+            return keycodeItem.code === code;
+        })[0];
+    }
+
+    env.ddg_spice_js_keycodes = function(api_result){
+        var query = DDG.get_query(),
+            key = query.replace(/js|javascript|key\s?(code)?s?/g, "").trim().toLowerCase(),
+            data = {},
+            $spice,
+            $table,
+            $input,
+            $result;
+
+        // if the query was something like "keycode tab"
+        // display the result for that
+        // otherwise we render the keycode tester
+        if (key) {
+            data.result = getKeycodeByKey(key);
+        }
+
+        // grab all the codes for the keycodes table
+        // manually add the index as this version of handlebars
+        // doesn't support @index inside #each
+        data.codes = $.map(keycodeLookup, function (keycode, index) {
+            return {
+                key: keycode.key,
+                code: keycode.code,
+                index: index
+            };
+        });
+
+        Spice.registerHelper({
+            // output three keycodes per row
+            tr: function (options) {
+                var markup = "";
+                if (this.index % 3 === 0) {
+                    markup += "<tr>";
+                }
+
+                markup += options.fn(this);
+
+                if (this.index % 3 === 2) {
+                    markup += "</tr>";
+                }
+
+                return markup;
+            },
+            // add separator between keycodes
+            separator: function () {
+                if (this.index % 3 !== 2) {
+                    return "class='with-separator'";
+                }
+            }
+        });
+
+        Spice.add({
+            id: "js_keycodes",
+            name: "Keycodes",
+            data: data,
+            meta: {},
+            templates: {
+                group: 'base',
+                options: {
+                    content: Spice.js_keycodes.content
+                }
```

Comment:
I would suggest you put the JS below into an `onShow` [event](https://duck.co/duckduckhack/spice_displaying#onshow-codefunctioncode)